handbook.md

```diff
@@ -0,0 +1,233 @@
+
+* **release (i.e. 1.10)**
+    * Code is cherry-picked using `git cherry-pick -m` into this tag.
+    * It contains the latest stable code for a *specific release*, for instance, 1.10.0.
+
+We do not force push to any of these branches. If a commit must be reverted, use `git-revert`, and push the reverted commit.
+
+Foreman projects strive to have a linear history ([like this](http://i.imgur.com/RMFS3bt.png)). A way to achieve this is to never create merge commits, by always rebasing your changes with develop `git pull --rebase upstream develop`. This allows us to easily revert our code to any point in time, which is helpful to detect in what commit we introduced an error using `git-bisect`.
+
+## Changelog
+
+TODO - Try to describe how this looks https://www.mozilla.org/en-US/firefox/39.0/releasenotes/
+
+## Coding standards
+
+The purpose of the Foreman coding standards is to create a baseline for collaboration and review within various aspects of the Foreman project and community, from core code to plugins.
+
+Coding standards help avoid common coding errors, improve the readability of code, and simplify modification. They ensure that files within the project appear as if they were created by a single person.
+
+### Pull requests
+
+All pull requests need to have an associated issue in the Foreman [issue tracker](http://projects.theforeman.org/).
+
+Foreman's [PR processor](https://github.com/theforeman/prprocessor) will parse all pull requests, assign labels, and run tests for all major projects.
+Pull requests are always *rebased* on top of the develop branch so that the git log stays linear.
+
+### Commit messages
+
+Provide a brief description of the change in the first line (50 chars or less), including a issue number.
+
+The title must start with ```Fixes #xxxx```. If your fix essentially extends a previous fix, use ```Refs #xxxx```. Our PR processor will understand these formats and will auto associate the pull request with the issue in the tracker.
+
+You can use the following model:
+
+```
+Fixes #9999 - lunch mars probe in V2 API.
+
+Some collaboration between teams will be necessary to accomplish this
+task. Other features, #2932 and #2933 will be closed after this. V1
+changes to the API were not necessary.
+
+
+```
+
+Insert a single blank line after the first line.
+
+Optionally, include more detailed explanatory text if necessary and wrap it to about 72 characters or so. ```git commit``` automatically wraps the text to these limits as you type in.
+
+Ensure the description is of the change and not the bug title, e.g. "X now accepts Y when doing Z" rather than "Z throws error"
+
+####Notes
+* Some background to justify this commit message style can be found [here](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+* By adding 'Refs #<issue number>' PR processor will auto add the commit to an existing issue. Usually an already closed issue, or just to add some code to a existing issue with another PR open.
```

Comment:
I think @domcleal mentioned this elsewhere, but we don't use #Refs for closed issues right?